miasm2/core/cpu.py

```diff
@@ -361,29 +254,187 @@ def neg_int(t):
 plusop = pyparsing.oneOf('+ -')
 
 
-def gen_base_expr():
-    variable = pyparsing.Word(pyparsing.alphas + "_$.",
-                              pyparsing.alphanums + "_")
-    variable.setParseAction(parse_id)
-    operand = str_int | variable
-    base_expr = pyparsing.operatorPrecedence(operand,
-                                   [("!", 1, pyparsing.opAssoc.RIGHT, parse_op),
-                                    (logicop, 2, pyparsing.opAssoc.RIGHT,
-                                     parse_op),
-                                    (signop, 1, pyparsing.opAssoc.RIGHT,
-                                     parse_op),
-                                    (multop, 2, pyparsing.opAssoc.LEFT,
-                                     parse_op),
-                                    (plusop, 2, pyparsing.opAssoc.LEFT,
-                                     parse_op),
-                                    ])
-    return variable, operand, base_expr
+##########################
+
+def literal_list(l):
+    l = l[:]
+    l.sort()
+    l = l[::-1]
+    o = pyparsing.Literal(l[0])
+    for x in l[1:]:
+        o |= pyparsing.Literal(x)
+    return o
+
+
+def cb_int(t):
+    assert len(t) == 1
+    integer = AstInt(t[0])
+    return integer
+
+
+def cb_parse_id(t):
+    assert len(t) == 1
+    reg = t[0]
+    return AstId(reg)
 
 
-variable, operand, base_expr = gen_base_expr()
+def cb_reg2exprid(t):
+    assert len(t) == 1
+    reg = t[0]
+    return AstId(reg)
+
+
+def cb_deref_reg(t):
+    assert len(t) == 1
+    reg = t[0]
+    ptr = reg
+    mem = AstMem(ptr, 32)
```

Comment:
Why all `AstMem` in these functions have a default size of 32?

Reply:
Correct: unused function: removed